services/pgnotifier/pgnotifier.go

```diff
@@ -272,10 +274,19 @@ func (notifier *PgNotifierT) updateClaimedEvent(id int64, ch chan ClaimResponseT
 									THEN CAST ( '%[3]s' AS pg_notifier_status_type)
 									ELSE  CAST( '%[4]s' AS pg_notifier_status_type)
 									END), attempt = attempt + 1, updated_at = '%[5]s', error = %[6]s
-									WHERE id = %[7]v`, queueName, maxAttempt, AbortedState, FailedState, GetCurrentSQLTimestamp(), misc.QuoteLiteral(response.Err.Error()), id)
+									WHERE id = %[7]v`, queueName, maxAttempt, AbortedState, FailedState, GetCurrentSQLTimestamp(), misc.QuoteLiteral(response.Err.Error()), claim.ID)
 			_, err = notifier.dbHandle.Exec(stmt)
+
+			// Sending stats when we mark pg_notifier status as aborted.
+			if claim.Attempt > maxAttempt {
```

Comment:
don't we need to increment claim.Attempt somewhere? It is set at claim time and not incremented anywhere else

Reply:
Got resolved.